File: MouseDrag.js
#pragma strict

var distance : float = 10;

function OnMouseDrag () {

    if (Input.GetAxis("Mouse ScrollWheel") > 0) // forward
    {
        distance++;
    }
    if (Input.GetAxis("Mouse ScrollWheel") < 0) // back
    {
        distance--;
    }

    var mousePosition : Vector3 = new Vector3(Input.mousePosition.x, Input.mousePosition.y, distance);
    var objPosition : Vector3 = Camera.main.ScreenToWorldPoint(mousePosition);

    transform.position = objPosition;
}

File: mouseDrag.cs
using UnityEngine;
using System.Collections;
using UnityEngine.UI;

public class mouseDrag : MonoBehaviour {
	
	public Text ResultText;
	// Use this for initialization
	float distance = 10;
	public GameObject fireoff;

	void start(){
		ResultText.text = "";
	}

	void OnMouseDrag()
	{
		//Debug.Log ("Mouse drag");
		Vector3 mousePosition = new Vector3 (Input.mousePosition.x, Input.mousePosition.y, distance);
		Vector3 objPosition = Camera.main.ScreenToWorldPoint (mousePosition);

		transform.position = objPosition;


	}
	void OnTriggerEnter(Collider other){
		if (other.CompareTag("element")) {
			ResultText.text = "fire is formed ";
			//DestroyObject (other);

			Destroy (gameObject);
			//gameObject.fireoff.SetActive (false);

			fireoff.SetActive (true);

		}

	}
}
